// @dhis2/app-service-data/build/cjs/react/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CustomDataProvider", {
  enumerable: true,
  get: function () {
    return _CustomDataProvider.CustomDataProvider;
  }
});
Object.defineProperty(exports, "DataMutation", {
  enumerable: true,
  get: function () {
    return _DataMutation.DataMutation;
  }
});
Object.defineProperty(exports, "DataProvider", {
  enumerable: true,
  get: function () {
    return _DataProvider.DataProvider;
  }
});
Object.defineProperty(exports, "DataQuery", {
  enumerable: true,
  get: function () {
    return _DataQuery.DataQuery;
  }
});
Object.defineProperty(exports, "DataContext", {
  enumerable: true,
  get: function () {
    return _DataContext.DataContext;
  }
});
Object.defineProperty(exports, "defaultContext", {
  enumerable: true,
  get: function () {
    return _defaultContext.defaultContext;
  }
});
Object.defineProperty(exports, "useDataEngine", {
  enumerable: true,
  get: function () {
    return _useDataEngine.useDataEngine;
  }
});
Object.defineProperty(exports, "useDataMutation", {
  enumerable: true,
  get: function () {
    return _useDataMutation.useDataMutation;
  }
});
Object.defineProperty(exports, "useDataQuery", {
  enumerable: true,
  get: function () {
    return _useDataQuery.useDataQuery;
  }
});
Object.defineProperty(exports, "useQueryExecutor", {
  enumerable: true,
  get: function () {
    return _useQueryExecutor.useQueryExecutor;
  }
});
Object.defineProperty(exports, "useStaticInput", {
  enumerable: true,
  get: function () {
    return _useStaticInput.useStaticInput;
  }
});

var _CustomDataProvider = require("./components/CustomDataProvider");

var _DataMutation = require("./components/DataMutation");

var _DataProvider = require("./components/DataProvider");

var _DataQuery = require("./components/DataQuery");

var _DataContext = require("./context/DataContext");

var _defaultContext = require("./context/defaultContext");

var _useDataEngine = require("./hooks/useDataEngine");

var _useDataMutation = require("./hooks/useDataMutation");

var _useDataQuery = require("./hooks/useDataQuery");

var _useQueryExecutor = require("./hooks/useQueryExecutor");

var _useStaticInput = require("./hooks/useStaticInput");
