// @dhis2/app-service-data/build/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "DataQuery", {
  enumerable: true,
  get: function () {
    return _react.DataQuery;
  }
});
Object.defineProperty(exports, "DataMutation", {
  enumerable: true,
  get: function () {
    return _react.DataMutation;
  }
});
Object.defineProperty(exports, "DataProvider", {
  enumerable: true,
  get: function () {
    return _react.DataProvider;
  }
});
Object.defineProperty(exports, "CustomDataProvider", {
  enumerable: true,
  get: function () {
    return _react.CustomDataProvider;
  }
});
Object.defineProperty(exports, "useDataEngine", {
  enumerable: true,
  get: function () {
    return _react.useDataEngine;
  }
});
Object.defineProperty(exports, "useDataQuery", {
  enumerable: true,
  get: function () {
    return _react.useDataQuery;
  }
});
Object.defineProperty(exports, "useDataMutation", {
  enumerable: true,
  get: function () {
    return _react.useDataMutation;
  }
});

var _react = require("./react");
