// @dhis2-ui/selector-bar/build/es/selector-bar/selector-bar.js
import _JSXStyle from "styled-jsx/style";
import { Button } from '@dhis2-ui/button';
import i18n from '@dhis2/d2-i18n';
import { colors, spacers } from '@dhis2/ui-constants';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const ClearSelection = _ref => {
  let {
    disabled,
    onClick
  } = _ref;
  return /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["3676026375", [spacers.dp8]]]) + " " + "clear-selections"
  }, /*#__PURE__*/React.createElement(Button, {
    small: true,
    secondary: true,
    disabled: disabled,
    onClick: onClick
  }, i18n.t('Clear selections')), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "3676026375",
    dynamic: [spacers.dp8]
  }, [".clear-selections.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:40px;padding:0 ".concat(spacers.dp8, ";}")]));
};

ClearSelection.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func
};
export const SelectorBar = _ref2 => {
  let {
    className,
    children,
    onClearSelectionClick,
    dataTest,
    disableClearSelections,
    additionalContent
  } = _ref2;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    "data-test": dataTest,
    className: _JSXStyle.dynamic([["720742913", [colors.white, colors.grey400,
    /*
     * Specs define the space to be 12px.
     * 8px already come from the clear selection component
     */
    '']]]) + " " + (cx('container', {
      withRHSContents: additionalContent
    }, className) || "")
  }, /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["720742913", [colors.white, colors.grey400, '']]]) + " " + "controls"
  }, children, onClearSelectionClick && /*#__PURE__*/React.createElement(ClearSelection, {
    disabled: disableClearSelections,
    onClick: onClearSelectionClick
  })), additionalContent && /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["720742913", [colors.white, colors.grey400, '']]]) + " " + "additional-contents"
  }, additionalContent), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "720742913",
    dynamic: [colors.white, colors.grey400, '']
  }, [".container.__jsx-style-dynamic-selector{background:".concat(colors.white, ";box-shadow:inset 0 -1px 0 0 ").concat(colors.grey400, ";padding-bottom:1px;}"), ".withRHSContents.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".controls.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:1px;}", ".additional-contents.__jsx-style-dynamic-selector{".concat(" padding-left:4px;}")])));
};
SelectorBar.defaultProps = {
  dataTest: 'dhis2-ui-selectorbar'
};
SelectorBar.propTypes = {
  children: PropTypes.any.isRequired,
  additionalContent: PropTypes.any,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  disableClearSelections: PropTypes.bool,
  onClearSelectionClick: PropTypes.func
};
