// @dhis2/ui-forms/build/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  FinalForm: true,
  ReactFinalForm: true,
  CheckboxFieldFF: true,
  FileInputFieldFF: true,
  InputFieldFF: true,
  MultiSelectFieldFF: true,
  SingleSelectFieldFF: true,
  RadioFieldFF: true,
  SwitchFieldFF: true,
  TextAreaFieldFF: true,
  FieldGroupFF: true
};
Object.defineProperty(exports, "CheckboxFieldFF", {
  enumerable: true,
  get: function () {
    return _CheckboxFieldFF.CheckboxFieldFF;
  }
});
Object.defineProperty(exports, "FieldGroupFF", {
  enumerable: true,
  get: function () {
    return _FieldGroupFF.FieldGroupFF;
  }
});
Object.defineProperty(exports, "FileInputFieldFF", {
  enumerable: true,
  get: function () {
    return _FileInputFieldFF.FileInputFieldFF;
  }
});
exports.FinalForm = void 0;
Object.defineProperty(exports, "InputFieldFF", {
  enumerable: true,
  get: function () {
    return _InputFieldFF.InputFieldFF;
  }
});
Object.defineProperty(exports, "MultiSelectFieldFF", {
  enumerable: true,
  get: function () {
    return _MultiSelectFieldFF.MultiSelectFieldFF;
  }
});
Object.defineProperty(exports, "RadioFieldFF", {
  enumerable: true,
  get: function () {
    return _RadioFieldFF.RadioFieldFF;
  }
});
exports.ReactFinalForm = void 0;
Object.defineProperty(exports, "SingleSelectFieldFF", {
  enumerable: true,
  get: function () {
    return _SingleSelectFieldFF.SingleSelectFieldFF;
  }
});
Object.defineProperty(exports, "SwitchFieldFF", {
  enumerable: true,
  get: function () {
    return _SwitchFieldFF.SwitchFieldFF;
  }
});
Object.defineProperty(exports, "TextAreaFieldFF", {
  enumerable: true,
  get: function () {
    return _TextAreaFieldFF.TextAreaFieldFF;
  }
});

var FinalForm = _interopRequireWildcard(require("final-form"));

exports.FinalForm = FinalForm;

var ReactFinalForm = _interopRequireWildcard(require("react-final-form"));

exports.ReactFinalForm = ReactFinalForm;

var _CheckboxFieldFF = require("./CheckboxFieldFF/CheckboxFieldFF.js");

var _FileInputFieldFF = require("./FileInputFieldFF/FileInputFieldFF.js");

var _InputFieldFF = require("./InputFieldFF/InputFieldFF.js");

var _MultiSelectFieldFF = require("./MultiSelectFieldFF/MultiSelectFieldFF.js");

var _SingleSelectFieldFF = require("./SingleSelectFieldFF/SingleSelectFieldFF.js");

var _RadioFieldFF = require("./RadioFieldFF/RadioFieldFF.js");

var _SwitchFieldFF = require("./SwitchFieldFF/SwitchFieldFF.js");

var _TextAreaFieldFF = require("./TextAreaFieldFF/TextAreaFieldFF.js");

var _FieldGroupFF = require("./FieldGroupFF/FieldGroupFF.js");

var _index = require("./transformers/index.js");

Object.keys(_index).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _index[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _index[key];
    }
  });
});

var _index2 = require("./validators/index.js");

Object.keys(_index2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _index2[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _index2[key];
    }
  });
});

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
