// @dhis2-ui/segmented-control/build/cjs/segmented-control.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.DisabledOption = exports.Default = void 0;

var _react = _interopRequireDefault(require("react"));

var _segmentedControl = require("./segmented-control.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const logger = _ref => {
  let {
    value
  } = _ref;
  return console.log("Selected value: ".concat(value));
};

var _default = {
  title: 'Segmented Control',
  component: _segmentedControl.SegmentedControl,
  parameters: {
    componentSubtitle: 'Allows selection between related options'
  },
  args: {
    options: [{
      label: 'Dog',
      value: 'DOG'
    }, {
      label: 'Cat',
      value: 'CAT'
    }, {
      label: 'Giraffe',
      value: 'GIRAFFE'
    }],
    selected: 'DOG',
    onChange: logger
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_segmentedControl.SegmentedControl, args);

const Default = Template.bind({});
exports.Default = Default;
const DisabledOption = Template.bind({});
exports.DisabledOption = DisabledOption;
DisabledOption.args = {
  options: [{
    label: 'One',
    value: 'ONE'
  }, {
    label: 'Two',
    value: 'TWO',
    disabled: true
  }, {
    label: 'Three',
    value: 'THREE'
  }],
  selected: 'THREE'
};
