// @dhis2/app-service-offline/build/cjs/lib/cacheable-section.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useCacheableSection = useCacheableSection;
exports.CacheableSection = CacheableSection;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _cacheableSectionState = require("./cacheable-section-state");

var _offlineInterface = require("./offline-interface");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const recordingStates = {
  default: 'default',
  pending: 'pending',
  recording: 'recording',
  error: 'error'
};

/**
 * Returns the main controls for a cacheable section and manages recording
 * state, which affects the render state of the CacheableSection component.
 * Also returns the cached status of the section, which come straight from
 * the `useCachedSection` hook.
 *
 * @param {String} id
 * @returns {Object}
 */
function useCacheableSection(id) {
  const offlineInterface = (0, _offlineInterface.useOfflineInterface)();
  const {
    isCached,
    lastUpdated,
    remove,
    syncCachedSections
  } = (0, _cacheableSectionState.useCachedSection)(id);
  const {
    recordingState,
    setRecordingState,
    removeRecordingState
  } = (0, _cacheableSectionState.useRecordingState)(id);
  (0, _react.useEffect)(() => {
    // On mount, add recording state for this ID to context if needed
    if (!recordingState) {
      setRecordingState(recordingStates.default);
    } // On unnmount, remove recording state if not recording


    return () => {
      if (recordingState && recordingState !== recordingStates.recording && recordingState !== recordingStates.pending) {
        removeRecordingState();
      }
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  function startRecording({
    recordingTimeoutDelay = 1000,
    onStarted,
    onCompleted,
    onError
  } = {}) {
    // This promise resolving means that the message to the service worker
    // to start recording was successful. Waiting for resolution prevents
    // unnecessarily rerendering the whole component in case of an error
    return offlineInterface.startRecording({
      sectionId: id,
      recordingTimeoutDelay,
      onStarted: () => {
        onRecordingStarted();
        onStarted && onStarted();
      },
      onCompleted: () => {
        onRecordingCompleted();
        onCompleted && onCompleted();
      },
      onError: error => {
        onRecordingError(error);
        onError && onError(error);
      }
    }).then(() => setRecordingState(recordingStates.pending));
  }

  function onRecordingStarted() {
    setRecordingState(recordingStates.recording);
  }

  function onRecordingCompleted() {
    setRecordingState(recordingStates.default);
    syncCachedSections();
  }

  function onRecordingError(error) {
    console.error('Error during recording:', error);
    setRecordingState(recordingStates.error);
  } // isCached, lastUpdated, remove: _could_ be accessed by useCachedSection,
  // but provided through this hook for convenience


  return {
    recordingState,
    startRecording,
    lastUpdated,
    isCached,
    remove
  };
}

/**
 * Used to wrap the relevant component to be recorded and saved offline.
 * Depending on the recording state of the section, this wrapper will
 * render its children, not render its children while recording is pending,
 * or RErerender the chilren to force data fetching to record by the service
 * worker.
 *
 * During recording, a loading mask provided by props is also rendered that is
 * intended to prevent other interaction with the app that might interfere
 * with the recording process.
 */
function CacheableSection({
  id,
  loadingMask,
  children
}) {
  // Accesses recording state that useCacheableSection controls
  const {
    recordingState
  } = (0, _cacheableSectionState.useRecordingState)(id); // The following causes the component to reload in the event of a recording
  // error; the state will be cleared next time recording moves to pending.
  // It fixes a component getting stuck while rendered without data after
  // failing a recording while offline.
  // Errors can be handled in the `onError` callback to `startRecording`.

  if (recordingState === recordingStates.error) {
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, children);
  } // Handling rendering with the following conditions prevents an unncessary
  // rerender after successful recording


  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, recordingState === recordingStates.recording && loadingMask, recordingState !== recordingStates.pending && children);
}

CacheableSection.propTypes = {
  id: _propTypes.default.string.isRequired,
  children: _propTypes.default.node,
  loadingMask: _propTypes.default.node
};
