// @dhis2-ui/sharing-dialog/build/cjs/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VISUALIZATION = exports.SHARE_TARGET_USER = exports.SHARE_TARGET_PUBLIC = exports.SHARE_TARGET_GROUP = exports.SHARE_TARGET_EXTERNAL = exports.INTERPRETATION = exports.EVENT_VISUALIZATION = exports.DASHBOARD = exports.ACCESS_VIEW_ONLY = exports.ACCESS_VIEW_AND_EDIT = exports.ACCESS_NONE = void 0;

/**
 * Access types
 */
const ACCESS_NONE = 'ACCESS_NONE';
exports.ACCESS_NONE = ACCESS_NONE;
const ACCESS_VIEW_ONLY = 'ACCESS_VIEW_ONLY';
exports.ACCESS_VIEW_ONLY = ACCESS_VIEW_ONLY;
const ACCESS_VIEW_AND_EDIT = 'ACCESS_VIEW_AND_EDIT';
/**
 * Sharing targets
 */

exports.ACCESS_VIEW_AND_EDIT = ACCESS_VIEW_AND_EDIT;
const SHARE_TARGET_EXTERNAL = 'SHARE_TARGET_EXTERNAL';
exports.SHARE_TARGET_EXTERNAL = SHARE_TARGET_EXTERNAL;
const SHARE_TARGET_PUBLIC = 'SHARE_TARGET_PUBLIC';
exports.SHARE_TARGET_PUBLIC = SHARE_TARGET_PUBLIC;
const SHARE_TARGET_USER = 'SHARE_TARGET_USER';
exports.SHARE_TARGET_USER = SHARE_TARGET_USER;
const SHARE_TARGET_GROUP = 'SHARE_TARGET_GROUP';
/**
 * Sharing dialog types
 */

exports.SHARE_TARGET_GROUP = SHARE_TARGET_GROUP;
const VISUALIZATION = 'visualization';
exports.VISUALIZATION = VISUALIZATION;
const DASHBOARD = 'dashboard';
exports.DASHBOARD = DASHBOARD;
const EVENT_VISUALIZATION = 'eventVisualization';
exports.EVENT_VISUALIZATION = EVENT_VISUALIZATION;
const INTERPRETATION = 'interpretation';
exports.INTERPRETATION = INTERPRETATION;
