// @dhis2-ui/organisation-unit-tree/build/cjs/__stories__/force-reload-all.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ForceReloadAll = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _ui = require("@dhis2/ui");

var _react = _interopRequireWildcard(require("react"));

var _index = require("../index.js");

var _shared = require("./shared.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ForceReloadAll = () => {
  const [forceReload, _setForceReload] = (0, _react.useState)(false);

  const setForceReload = v => console.log('setForceReload', v) || _setForceReload(v);

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_ui.Button, {
    disabled: forceReload,
    onClick: () => setForceReload(true)
  }, "Reload org unit tree"), ' ', /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-2109967648"
  }, "(Force reload: ", forceReload ? 'true' : 'false', ")"), /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-2109967648"
  }, /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
    onChange: _shared.onChange,
    forceReload: forceReload,
    name: "Root org unit",
    roots: ['A0000000000'],
    initiallyExpanded: ['/A0000000000/A0000000001'],
    selected: ['/A0000000000/A0000000001/A0000000003'],
    onChildrenLoaded: data => {
      const {
        id
      } = data;

      if (id === 'A0000000000') {
        setForceReload(false);
      }
    }
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2109967648"
  }, ["div.jsx-2109967648{width:400px;border:1px solid black;margin-top:32px;padding:16px;min-height:200px;}"]));
};

exports.ForceReloadAll = ForceReloadAll;
ForceReloadAll.storyName = 'Force reload all';
