// @dhis2/ui-forms/build/es/SwitchFieldFF/SwitchFieldFF.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { SwitchField } from '@dhis2-ui/switch';
import PropTypes from 'prop-types';
import React from 'react';
import { createToggleChangeHandler, createFocusHandler, createBlurHandler, hasError, isValid, getValidationText } from '../shared/helpers.js';
import { metaPropType, inputPropType } from '../shared/propTypes.js';
export const SwitchFieldFF = _ref => {
  let {
    error,
    input,
    meta,
    showValidStatus,
    valid,
    validationText,
    onBlur,
    onFocus,
    ...rest
  } = _ref;
  return /*#__PURE__*/React.createElement(SwitchField, _extends({}, rest, {
    checked: input.checked,
    name: input.name,
    value: input.value,
    error: hasError(meta, error),
    valid: isValid(meta, valid, showValidStatus),
    validationText: getValidationText(meta, validationText, error),
    onFocus: createFocusHandler(input, onFocus),
    onChange: createToggleChangeHandler(input),
    onBlur: createBlurHandler(input, onBlur)
  }));
};
SwitchFieldFF.propTypes = {
  /** `input` props received from Final Form `Field` */
  input: inputPropType.isRequired,

  /** `meta` props received from Final Form `Field` */
  meta: metaPropType.isRequired,
  error: PropTypes.bool,
  showValidStatus: PropTypes.bool,
  valid: PropTypes.bool,
  validationText: PropTypes.string,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func
};
