// @dhis2-ui/center/build/cjs/center.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Top = exports.Default = exports.Bottom = void 0;

var _react = _interopRequireDefault(require("react"));

var _center = require("./center.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = "\nCenters children horizontally, and by default, vertically.\nUse the `top` or `bottom` props to change vertical alignment.\n\n```js\nimport { CenteredContent } from '@dhis2/ui'\n```\n";

const Wrapper = story => /*#__PURE__*/_react.default.createElement("div", {
  style: {
    height: '150px'
  }
}, story());

var _default = {
  title: 'Center',
  component: _center.Center,
  decorators: [Wrapper],
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_center.Center, args, /*#__PURE__*/_react.default.createElement("span", null, "Center me"));

const Default = Template.bind({});
exports.Default = Default;
const Top = Template.bind({});
exports.Top = Top;
Top.args = {
  position: 'top'
};
const Bottom = Template.bind({});
exports.Bottom = Bottom;
Bottom.args = {
  position: 'bottom'
};
