// @dhis2-ui/user-avatar/build/cjs/user-avatar.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Sizes = exports.Default = void 0;

var _react = _interopRequireDefault(require("react"));

var _userAvatar = require("./user-avatar.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'User Avatar',
  component: _userAvatar.UserAvatar
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, args);

const Default = Template.bind({});
exports.Default = Default;
Default.args = {
  name: 'John Doe'
};

const Sizes = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, {
  name: "John Doe",
  extrasmall: true
}), /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, {
  name: "John Doe",
  small: true
}), /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, {
  name: "John Doe",
  medium: true
}), /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, {
  name: "John Doe",
  large: true
}), /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, {
  name: "John Doe",
  extralarge: true
}));

exports.Sizes = Sizes;
