// @dhis2-ui/organisation-unit-tree/build/es/organisation-unit-node/label/label-container.js
import _JSXStyle from "styled-jsx/style";
import { colors } from '@dhis2/ui-constants';
import cx from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
/**
 * @param {Object} props
 * @param {bool} props.highlighted
 * @param {React.Component|React.Component[]} props.children
 * @returns {React.Component}
 */

export const LabelContainer = _ref => {
  let {
    highlighted,
    children
  } = _ref;
  return /*#__PURE__*/React.createElement("div", {
    className: _JSXStyle.dynamic([["495975020", [colors.teal200]]]) + " " + (cx({
      highlighted
    }) || "")
  }, /*#__PURE__*/React.createElement("span", {
    className: _JSXStyle.dynamic([["495975020", [colors.teal200]]])
  }, children), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "495975020",
    dynamic: [colors.teal200]
  }, ["div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", "span.__jsx-style-dynamic-selector{display:block;}", ".highlighted.__jsx-style-dynamic-selector{background:".concat(colors.teal200, ";padding-right:4px;}")]));
};
LabelContainer.propTypes = {
  children: PropTypes.node,
  highlighted: PropTypes.bool
};
