// @dhis2-ui/modal/build/cjs/modal-actions/modal-actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalActions = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _uiConstants = require("@dhis2/ui-constants");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ModalActions = _ref => {
  let {
    children,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: _style.default.dynamic([["2674552674", [_uiConstants.spacers.dp16]]])
  }, children, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2674552674",
    dynamic: [_uiConstants.spacers.dp16]
  }, ["div.__jsx-style-dynamic-selector{-webkit-order:3;-ms-flex-order:3;order:3;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;-webkit-align-self:flex-end;-ms-flex-item-align:end;align-self:flex-end;margin:".concat(_uiConstants.spacers.dp16, " 0 0;}")]));
};

exports.ModalActions = ModalActions;
ModalActions.defaultProps = {
  dataTest: 'dhis2-uicore-modalactions'
};
ModalActions.propTypes = {
  children: _propTypes.default.node,
  dataTest: _propTypes.default.string
};
