// @dhis2-ui/table/build/es/table/table-row.js
import _JSXStyle from "styled-jsx/style";
import cx from 'classnames';
import PropTypes from 'prop-types';
import React, { useContext } from 'react';
import { TableContext } from './table-context.js';
const tableRowStyles = [".zebraStriping.jsx-3124144591:nth-child(even){background:#fbfcfd;}"];
tableRowStyles.__hash = "3124144591";
export const TableRow = _ref => {
  let {
    role,
    children,
    className,
    dataTest,
    suppressZebraStriping
  } = _ref;
  const {
    suppressZebraStriping: suppressZebraStripingFromContext
  } = useContext(TableContext);
  const zebraStriping = typeof suppressZebraStriping !== 'undefined' ? !suppressZebraStriping : !suppressZebraStripingFromContext;
  return /*#__PURE__*/React.createElement("tr", {
    "data-test": dataTest,
    role: role,
    className: "jsx-".concat(tableRowStyles.__hash) + " " + (cx(className, {
      zebraStriping
    }) || "")
  }, children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: tableRowStyles.__hash
  }, tableRowStyles));
};
TableRow.defaultProps = {
  dataTest: 'dhis2-uicore-tablerow'
};
TableRow.propTypes = {
  /** Should be `<TableCell>` or `<TableCellHead>` components */
  children: PropTypes.node,
  className: PropTypes.string,
  dataTest: PropTypes.string,
  role: PropTypes.string,

  /** Disables the default row striping for this row */
  suppressZebraStriping: PropTypes.bool
};
