// @dhis2-ui/css/build/es/css-variables/css-variables.stories.js
import React from 'react';
import { CssVariables } from './index.js';
const description = "\nA utility for adding DHIS2 theme variables to global CSS variables.\n\n```js\nimport { CssVariables } from '@dhis2/ui'\n```\n";

const App = _ref => {
  let {
    children
  } = _ref;
  return /*#__PURE__*/React.createElement("div", null, children);
};

export default {
  title: 'CSS Variables',
  component: CssVariables,
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  }
};
export const AllVariables = args => /*#__PURE__*/React.createElement(App, null, /*#__PURE__*/React.createElement(CssVariables, args), /*#__PURE__*/React.createElement("p", null, "The sections of the theme that should be inserted can be toggled with flags, which allows the theme variables to be used as regular CSS custom properties. So this", ' ', /*#__PURE__*/React.createElement("span", {
  style: {
    color: 'var(--colors-red500)'
  }
}, "text"), " uses the vanilla CSS", ' ', /*#__PURE__*/React.createElement("span", {
  style: {
    color: 'var(--colors-blue500)'
  }
}, "custom properties"), ' ', "set by the CssVariables component."));
AllVariables.args = {
  colors: true,
  theme: true,
  layers: true,
  spacers: true,
  elevations: true
};
export const NoVariables = args => /*#__PURE__*/React.createElement(App, null, /*#__PURE__*/React.createElement(CssVariables, args), /*#__PURE__*/React.createElement("p", null, "By default no custom properties are inserted."));
