// @dhis2/ui-forms/build/cjs/InputFieldFF/InputFieldFF.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Statuses = exports.Required = exports.HelpText = exports.Disabled = exports.Default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactFinalForm = require("react-final-form");

var _formDecorator = require("../formDecorator.js");

var _propTypes = require("../shared/propTypes.js");

var _index = require("../validators/index.js");

var _InputFieldFF = require("./InputFieldFF.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const description = "\nThe `InputFieldFF` is a wrapper around a `InputField` that enables it to work with Final Form, the preferred library for form validation and utilities in DHIS 2 apps.\n\n#### Final Form\n\nSee how to use Final Form at [Final Form - Getting Started](https://final-form.org/docs/react-final-form/getting-started).\n\nInside a Final Form `<Form>` component, these 'FF' UI components are intended to be used in the `component` prop of the [Final Form `<Field>` components](https://final-form.org/docs/react-final-form/api/Field) where they will receive some props from the Field, e.g. `<Field component={InputFieldFF} />`.  See the code samples below for examples.\n\n#### Props\n\nThe props shown in the table below are generally provided to the `InputFieldFF` wrapper by the Final Form `Field`.\n\nNote that any props beyond the API of the `Field` component will be spread to the `InputFieldFF`, which passes any extra props to the underlying `InputField` using `{...rest}`.\n\nTherefore, to add any props to the `InputFieldFF` or `InputField`, add those props to the parent Final Form `Field` component.\n\nAlso see `InputField` for notes about props and implementation.\n\n```js\nimport { InputFieldFF } from '@dhis2/ui'\n```\n\nPress **Submit** to see the form values logged to the console.\n";
var _default = {
  title: 'Input Field (Final Form)',
  component: _InputFieldFF.InputFieldFF,
  decorators: [_formDecorator.formDecorator],
  parameters: {
    docs: {
      description: {
        component: description
      }
    }
  },
  argTypes: {
    input: { ..._propTypes.inputArgType
    },
    meta: { ..._propTypes.metaArgType
    }
  }
};
exports.default = _default;

const Default = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  component: _InputFieldFF.InputFieldFF,
  name: "agree",
  label: "Do you agree?"
});

exports.Default = Default;

const Required = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  name: "agree",
  component: _InputFieldFF.InputFieldFF,
  required: true,
  validate: _index.hasValue,
  label: "Do you agree?"
});

exports.Required = Required;

const Disabled = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  name: "agree",
  component: _InputFieldFF.InputFieldFF,
  disabled: true,
  label: "Do you agree?"
});

exports.Disabled = Disabled;

const HelpText = () => /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  name: "agree",
  component: _InputFieldFF.InputFieldFF,
  label: "Do you agree?",
  helpText: "Click to agree"
});

exports.HelpText = HelpText;
HelpText.storyName = 'Help text';

const Statuses = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  name: "valid",
  component: _InputFieldFF.InputFieldFF,
  label: "Valid",
  valid: true,
  validationText: "Validation text"
}), /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  name: "warning",
  component: _InputFieldFF.InputFieldFF,
  label: "Warning",
  warning: true,
  validationText: "Validation text"
}), /*#__PURE__*/_react.default.createElement(_reactFinalForm.Field, {
  name: "error",
  component: _InputFieldFF.InputFieldFF,
  label: "Error",
  error: true,
  validationText: "Validation text"
}));

exports.Statuses = Statuses;
