// @dhis2-ui/selector-bar/build/cjs/selector-bar/__stories__/with-some-inputs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithSomeInputs = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../../index.js");

var _common = require("./common.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const WithSomeInputs = (_, _ref) => {
  let {
    workflow,
    setWorkflow,
    workflowOpen,
    setWorkflowOpen,
    orgUnit,
    setOrgUnit,
    orgUnitOpen,
    setOrgUnitOpen
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_index.SelectorBar, {
    disableClearSelections: !workflow && !orgUnit,
    onClearSelectionClick: () => {
      setWorkflow(null);
      setOrgUnit(null);
    }
  }, /*#__PURE__*/_react.default.createElement(_index.SelectorBarItem, {
    disabled: false,
    label: "Workflow",
    value: workflow === null || workflow === void 0 ? void 0 : workflow.label,
    noValueMessage: "Choose a workflow",
    open: workflowOpen,
    setOpen: setWorkflowOpen
  }, /*#__PURE__*/_react.default.createElement(_common.MenuSelect, {
    values: _common.workflows,
    selected: workflow === null || workflow === void 0 ? void 0 : workflow.value,
    onChange: _ref2 => {
      let {
        selected
      } = _ref2;
      setWorkflow(_common.workflows.find(currentWorkflow => selected === currentWorkflow.value));
      setWorkflowOpen(false);
    }
  })), /*#__PURE__*/_react.default.createElement(_index.SelectorBarItem, {
    disabled: !workflow,
    label: "Org unit",
    value: orgUnit === null || orgUnit === void 0 ? void 0 : orgUnit.displayName,
    noValueMessage: "Choose an organisation unit",
    open: orgUnitOpen,
    setOpen: setOrgUnitOpen
  }, /*#__PURE__*/_react.default.createElement(_common.OrgUnitSelect, {
    selected: orgUnit ? [orgUnit.path] : [],
    onChange: (nextOrgUnit, evt) => {
      evt.stopPropagation();
      setOrgUnit(nextOrgUnit);
      setOrgUnitOpen(false);
    }
  })));
};

exports.WithSomeInputs = WithSomeInputs;
WithSomeInputs.decorators = [_common.decoratorCommonStyles, (0, _common.createStatefulDecorator)({
  workflow: _common.workflows[0]
}), (0, _common.createDecoratorCustomDataProvider)()];
