// @dhis2-ui/header-bar/build/cjs/apps.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _card = require("@dhis2-ui/card");

var _input = require("@dhis2-ui/input");

var _appRuntime = require("@dhis2/app-runtime");

var _uiConstants = require("@dhis2/ui-constants");

var _uiIcons = require("@dhis2/ui-icons");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _joinPath = require("./join-path.js");

var _index = _interopRequireDefault(require("./locales/index.js"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Copied from here:
 * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Escaping
 */
function escapeRegExpCharacters(text) {
  return text.replace(/[/.*+?^${}()|[\]\\]/g, '\\$&');
}

function Search(_ref) {
  let {
    value,
    onChange
  } = _ref;
  const {
    baseUrl
  } = (0, _appRuntime.useConfig)();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "jsx-4264724627"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-4264724627"
  }, /*#__PURE__*/_react.default.createElement(_input.InputField, {
    value: value,
    name: "filter",
    placeholder: _index.default.t('Search apps'),
    onChange: onChange,
    initialFocus: true
  })), /*#__PURE__*/_react.default.createElement("span", {
    className: "jsx-4264724627"
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: (0, _joinPath.joinPath)(baseUrl, 'dhis-web-menu-management'),
    className: "jsx-4264724627"
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconSettings24, {
    color: _uiConstants.colors.grey700
  }))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4264724627"
  }, ["div.jsx-4264724627{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-wrap:nowrap;-ms-flex-wrap:nowrap;flex-wrap:nowrap;height:52px;margin:8px;}", "span.jsx-4264724627{-webkit-flex:1 100%;-ms-flex:1 100%;flex:1 100%;}", "span.jsx-4264724627:last-child{-webkit-flex:1 auto;-ms-flex:1 auto;flex:1 auto;margin:8px;}"]));
}

Search.propTypes = {
  value: _propTypes.default.string.isRequired,
  onChange: _propTypes.default.func.isRequired
};

function Item(_ref2) {
  let {
    name,
    path,
    img
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement("a", {
    href: path,
    className: _style.default.dynamic([["3670725638", [_uiConstants.theme.primary050]]])
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: img,
    alt: "app logo",
    className: _style.default.dynamic([["3670725638", [_uiConstants.theme.primary050]]])
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["3670725638", [_uiConstants.theme.primary050]]])
  }, name), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3670725638",
    dynamic: [_uiConstants.theme.primary050]
  }, ["a.__jsx-style-dynamic-selector{display:inline-block;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:96px;margin:8px;padding:8px;border-radius:12px;-webkit-text-decoration:none;text-decoration:none;cursor:pointer;}", "a.__jsx-style-dynamic-selector:hover,a.__jsx-style-dynamic-selector:focus{background-color:".concat(_uiConstants.theme.primary050, ";cursor:pointer;}"), "a.__jsx-style-dynamic-selector:hover>div.__jsx-style-dynamic-selector{font-weight:500;cursor:pointer;}", "img.__jsx-style-dynamic-selector{width:48px;height:48px;cursor:pointer;}", "div.__jsx-style-dynamic-selector{margin-top:14px;color:rgba(0,0,0,0.87);font-size:12px;-webkit-letter-spacing:0.01em;-moz-letter-spacing:0.01em;-ms-letter-spacing:0.01em;letter-spacing:0.01em;line-height:14px;text-align:center;cursor:pointer;}"]));
}

Item.propTypes = {
  img: _propTypes.default.string,
  name: _propTypes.default.string,
  path: _propTypes.default.string
};

function List(_ref3) {
  let {
    apps,
    filter
  } = _ref3;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "headerbar-apps-menu-list",
    className: "jsx-2076871745"
  }, apps.filter(_ref4 => {
    let {
      displayName,
      name
    } = _ref4;
    const appName = displayName || name;
    const formattedAppName = appName.toLowerCase();
    const formattedFilter = escapeRegExpCharacters(filter).toLowerCase();
    return filter.length > 0 ? formattedAppName.match(formattedFilter) : true;
  }).map((_ref5, idx) => {
    let {
      displayName,
      name,
      defaultAction,
      icon
    } = _ref5;
    return /*#__PURE__*/_react.default.createElement(Item, {
      key: "app-".concat(name, "-").concat(idx),
      name: displayName || name,
      path: defaultAction,
      img: icon
    });
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "2076871745"
  }, ["div.jsx-2076871745{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-align-content:flex-start;-ms-flex-line-pack:start;align-content:flex-start;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;width:30vw;min-width:300px;max-width:560px;min-height:200px;max-height:465px;margin:0 8px 8px 8px;overflow:auto;overflow-x:hidden;}"]));
}

List.propTypes = {
  apps: _propTypes.default.array,
  filter: _propTypes.default.string
};

const AppMenu = _ref6 => {
  let {
    apps,
    filter,
    onFilterChange
  } = _ref6;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": "headerbar-apps-menu",
    className: "jsx-3879001227"
  }, /*#__PURE__*/_react.default.createElement(_card.Card, null, /*#__PURE__*/_react.default.createElement(Search, {
    value: filter,
    onChange: onFilterChange
  }), /*#__PURE__*/_react.default.createElement(List, {
    apps: apps,
    filter: filter
  })), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3879001227"
  }, ["div.jsx-3879001227{z-index:10000;position:absolute;top:28px;right:-6px;border-top:4px solid transparent;}"]));
};

AppMenu.propTypes = {
  apps: _propTypes.default.array.isRequired,
  onFilterChange: _propTypes.default.func.isRequired,
  filter: _propTypes.default.string
};

const Apps = _ref7 => {
  let {
    apps
  } = _ref7;
  const [show, setShow] = (0, _react.useState)(false);
  const [filter, setFilter] = (0, _react.useState)('');
  const handleVisibilityToggle = (0, _react.useCallback)(() => setShow(!show), [show]);
  const handleFilterChange = (0, _react.useCallback)(_ref8 => {
    let {
      value
    } = _ref8;
    return setFilter(value);
  }, []);
  const containerEl = (0, _react.useRef)(null);
  const onDocClick = (0, _react.useCallback)(evt => {
    if (containerEl.current && !containerEl.current.contains(evt.target)) {
      setShow(false);
    }
  }, []);
  (0, _react.useEffect)(() => {
    document.addEventListener('click', onDocClick);
    return () => document.removeEventListener('click', onDocClick);
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: containerEl,
    "data-test": "headerbar-apps",
    className: "jsx-515098251"
  }, /*#__PURE__*/_react.default.createElement("button", {
    onClick: handleVisibilityToggle,
    "data-test": "headerbar-apps-icon",
    className: "jsx-515098251"
  }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconApps24, {
    color: _uiConstants.colors.white
  })), show ? /*#__PURE__*/_react.default.createElement(AppMenu, {
    apps: apps,
    filter: filter,
    onFilterChange: handleFilterChange
  }) : null, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "515098251"
  }, ["button.jsx-515098251{display:block;background:transparent;padding:0;border:0;cursor:pointer;}", "button.jsx-515098251:focus{outline:1px dotted white;}", "div.jsx-515098251{position:relative;width:24px;height:30px;margin:8px 0 0 0;}"]));
};

Apps.propTypes = {
  apps: _propTypes.default.array.isRequired
};
var _default = Apps;
exports.default = _default;
