// @dhis2-ui/header-bar/build/cjs/profile.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _userAvatar = require("@dhis2-ui/user-avatar");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _profileMenu = require("./profile/profile-menu.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class Profile extends _react.default.Component {
  constructor() {
    super(...arguments);

    _defineProperty(this, "state", {
      show: false
    });

    _defineProperty(this, "onDocClick", evt => {
      if (this.elContainer && !this.elContainer.contains(evt.target)) {
        this.setState({
          show: false
        });
      }
    });

    _defineProperty(this, "handleToggle", () => this.setState({
      show: !this.state.show
    }));
  }

  componentDidMount() {
    document.addEventListener('click', this.onDocClick);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.onDocClick);
  }

  render() {
    const {
      name,
      email,
      avatarId,
      helpUrl
    } = this.props;
    return /*#__PURE__*/_react.default.createElement("div", {
      ref: c => this.elContainer = c,
      "data-test": "headerbar-profile",
      className: "jsx-2642719072" + " " + "headerbar-profile"
    }, /*#__PURE__*/_react.default.createElement("button", {
      onClick: this.handleToggle,
      className: "jsx-2642719072" + " " + "headerbar-profile-btn"
    }, /*#__PURE__*/_react.default.createElement(_userAvatar.UserAvatar, {
      avatarId: avatarId,
      name: name,
      dataTest: "headerbar-profile-icon",
      medium: true
    })), this.state.show ? /*#__PURE__*/_react.default.createElement(_profileMenu.ProfileMenu, {
      avatarId: avatarId,
      name: name,
      email: email,
      helpUrl: helpUrl
    }) : null, /*#__PURE__*/_react.default.createElement(_style.default, {
      id: "2642719072"
    }, [".headerbar-profile.jsx-2642719072{position:relative;width:36px;height:36px;min-width:36px;min-height:36px;margin:2px 12px 0 24px;}", ".headerbar-profile-btn.jsx-2642719072{background:transparent;padding:0;border:0;cursor:pointer;width:100%;height:100%;}", ".headerbar-profile-btn.jsx-2642719072:focus{outline:1px dotted white;}"]));
  }

}

exports.default = Profile;
Profile.propTypes = {
  name: _propTypes.default.string.isRequired,
  avatarId: _propTypes.default.string,
  email: _propTypes.default.string,
  helpUrl: _propTypes.default.string
};
