// @dhis2/app-service-data/build/es/links/RestAPILink/queryToRequestOptions/multipartFormDataMatchers.js
/*
 * Requests that expect a "multipart/form-data" Content-Type have been collected by scanning
 * the developer documentation:
 * https://docs.dhis2.org/master/en/developer/html/dhis2_developer_manual_full.html
 */
// Post to 'dataValues' (send/update a data value; endpoint doesn't support JSON)
// For file-uploads too
export const isDataValue = (type, {
  resource
}) => type === 'create' && (resource === 'dataValues' || resource === 'dataValues/file'); // POST to 'fileResources' (upload a file resource)

export const isFileResourceUpload = (type, {
  resource
}) => type === 'create' && resource === 'fileResources'; // POST to 'messageConversations/attachments' (upload a message conversation attachment)

export const isMessageConversationAttachment = (type, {
  resource
}) => type === 'create' && resource === 'messageConversations/attachments'; // POST to `staticContent/${key}` (upload staticContent: logo_banner | logo_front)

export const isStaticContentUpload = (type, {
  resource
}) => {
  const pattern = /^staticContent\/(?:logo_banner|logo_front)$/;
  return type === 'create' && pattern.test(resource);
}; // POST to 'apps' (install an app)

export const isAppInstall = (type, {
  resource
}) => type === 'create' && resource === 'apps'; // POST to convert an SVG file

export const isSvgConversion = (type, {
  resource
}) => type === 'create' && (resource === 'svg.png' || resource === 'svg.pdf');
