// @dhis2/ui-constants/build/cjs/spacers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.spacersNum = exports.spacers = void 0;

/** Number values of the spacer definitions */
const spacersNum = {
  dp4: 4,
  dp8: 8,
  dp12: 12,
  dp16: 16,
  dp24: 24,
  dp32: 32,
  dp48: 48,
  dp64: 64,
  dp96: 96,
  dp128: 128,
  dp192: 192,
  dp256: 256,
  dp384: 384,
  dp512: 512,
  dp640: 640
};
/** Pixel values of the spacer definitions */

exports.spacersNum = spacersNum;
const spacers = {
  dp4: "".concat(spacersNum.dp4, "px"),
  dp8: "".concat(spacersNum.dp8, "px"),
  dp12: "".concat(spacersNum.dp12, "px"),
  dp16: "".concat(spacersNum.dp16, "px"),
  dp24: "".concat(spacersNum.dp24, "px"),
  dp32: "".concat(spacersNum.dp32, "px"),
  dp48: "".concat(spacersNum.dp48, "px"),
  dp64: "".concat(spacersNum.dp64, "px"),
  dp96: "".concat(spacersNum.dp96, "px"),
  dp128: "".concat(spacersNum.dp128, "px"),
  dp192: "".concat(spacersNum.dp192, "px"),
  dp256: "".concat(spacersNum.dp256, "px"),
  dp384: "".concat(spacersNum.dp384, "px"),
  dp512: "".concat(spacersNum.dp512, "px"),
  dp640: "".concat(spacersNum.dp640, "px")
};
exports.spacers = spacers;
