// @dhis2-ui/pagination/build/cjs/__fixtures__/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.noTotalAtLastPage = exports.noTotal = exports.atTenthPage = exports.atSecondPage = exports.atPageBeforeLast = exports.atLastPage = exports.atFirstPage = void 0;
const atFirstPage = {
  page: 1,
  pageCount: 21,
  total: 1035,
  pageSize: 50
};
exports.atFirstPage = atFirstPage;
const atSecondPage = {
  page: 2,
  pageCount: 21,
  total: 1035,
  pageSize: 50
};
exports.atSecondPage = atSecondPage;
const atTenthPage = {
  page: 10,
  pageCount: 21,
  total: 1035,
  pageSize: 50
};
exports.atTenthPage = atTenthPage;
const atLastPage = {
  page: 21,
  pageCount: 21,
  total: 1035,
  pageSize: 50
};
exports.atLastPage = atLastPage;
const atPageBeforeLast = {
  page: 20,
  pageCount: 21,
  total: 1035,
  pageSize: 50
};
exports.atPageBeforeLast = atPageBeforeLast;
const noTotal = {
  page: 3,
  pageSize: 50,
  isLastPage: false
};
exports.noTotal = noTotal;
const noTotalAtLastPage = {
  page: 4,
  pageSize: 50,
  isLastPage: true
};
exports.noTotalAtLastPage = noTotalAtLastPage;
