// @dhis2-ui/transfer/build/cjs/__e2e__/display-order.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SomeSelected = exports.NoSelection = void 0;

var _react = _interopRequireDefault(require("react"));

var _transfer = require("../transfer.js");

var _options = require("./common/options.js");

var _statefulDecorator = require("./common/stateful-decorator.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Transfer Display Order'
};
exports.default = _default;
window.options = _options.options;

const NoSelection = (_, _ref) => {
  let {
    selected,
    onChange
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    selected: selected,
    onChange: onChange,
    options: _options.options
  });
};

exports.NoSelection = NoSelection;

const SomeSelected = (_, _ref2) => {
  let {
    selected,
    onChange
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
    selected: selected,
    onChange: onChange,
    options: _options.options
  });
};

exports.SomeSelected = SomeSelected;
SomeSelected.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.slice(0, 4).map(_ref3 => {
      let {
        value
      } = _ref3;
      return value;
    })
  })]
};
