// @dhis2/app-service-data/build/cjs/react/hooks/useDataMutation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useDataMutation = void 0;

var _react = require("react");

var _useDataEngine = require("./useDataEngine");

var _useQueryExecutor = require("./useQueryExecutor");

var _useStaticInput = require("./useStaticInput");

const empty = {};

const useDataMutation = (mutation, {
  onComplete,
  onError,
  variables = empty,
  lazy = true
} = {}) => {
  const engine = (0, _useDataEngine.useDataEngine)();
  const [theMutation] = (0, _useStaticInput.useStaticInput)(mutation, {
    warn: true,
    name: 'mutation'
  });
  const execute = (0, _react.useCallback)(options => engine.mutate(theMutation, options), [engine, theMutation]);
  const {
    refetch: mutate,
    called,
    loading,
    error,
    data
  } = (0, _useQueryExecutor.useQueryExecutor)({
    execute,
    variables,
    singular: false,
    immediate: !lazy,
    onComplete,
    onError
  });
  return [mutate, {
    engine,
    called,
    loading,
    error,
    data
  }];
};

exports.useDataMutation = useDataMutation;
