// @dhis2-ui/notice-box/build/cjs/notice-box-title.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NoticeBoxTitle = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _uiConstants = require("@dhis2/ui-constants");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const NoticeBoxTitle = _ref => {
  let {
    title,
    dataTest
  } = _ref;

  if (!title) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement("h6", {
    "data-test": dataTest,
    className: _style.default.dynamic([["3611901971", [_uiConstants.colors.grey900, _uiConstants.spacers.dp12]]])
  }, title, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "3611901971",
    dynamic: [_uiConstants.colors.grey900, _uiConstants.spacers.dp12]
  }, ["h6.__jsx-style-dynamic-selector{color:".concat(_uiConstants.colors.grey900, ";font-size:14px;font-weight:500;line-height:20px;margin:0 0 ").concat(_uiConstants.spacers.dp12, " 0;}")]));
};

exports.NoticeBoxTitle = NoticeBoxTitle;
NoticeBoxTitle.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  title: _propTypes.default.string
};
