// @dhis2/app-service-offline/build/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OfflineProvider", {
  enumerable: true,
  get: function () {
    return _offlineProvider.OfflineProvider;
  }
});
Object.defineProperty(exports, "CacheableSection", {
  enumerable: true,
  get: function () {
    return _cacheableSection.CacheableSection;
  }
});
Object.defineProperty(exports, "useCacheableSection", {
  enumerable: true,
  get: function () {
    return _cacheableSection.useCacheableSection;
  }
});
Object.defineProperty(exports, "useCachedSections", {
  enumerable: true,
  get: function () {
    return _cacheableSectionState.useCachedSections;
  }
});
Object.defineProperty(exports, "useOnlineStatus", {
  enumerable: true,
  get: function () {
    return _onlineStatus.useOnlineStatus;
  }
});
Object.defineProperty(exports, "clearSensitiveCaches", {
  enumerable: true,
  get: function () {
    return _clearSensitiveCaches.clearSensitiveCaches;
  }
});

var _offlineProvider = require("./lib/offline-provider");

var _cacheableSection = require("./lib/cacheable-section");

var _cacheableSectionState = require("./lib/cacheable-section-state");

var _onlineStatus = require("./lib/online-status");

var _clearSensitiveCaches = require("./lib/clear-sensitive-caches");
