// @dhis2-ui/sharing-dialog/build/cjs/modal/modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = void 0;

var _button = require("@dhis2-ui/button");

var _modal = require("@dhis2-ui/modal");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _index = _interopRequireDefault(require("../locales/index.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Modal = _ref => {
  let {
    onClose,
    name,
    children
  } = _ref;
  const title = name ? _index.default.t('Sharing and access: {{- objectName}}', {
    objectName: name,
    nsSeparator: '|'
  }) : _index.default.t('Sharing and access');
  return /*#__PURE__*/_react.default.createElement(_modal.Modal, {
    large: true,
    position: "top",
    onClose: onClose
  }, /*#__PURE__*/_react.default.createElement(_modal.ModalTitle, null, title), /*#__PURE__*/_react.default.createElement(_modal.ModalContent, null, children), /*#__PURE__*/_react.default.createElement(_modal.ModalActions, null, /*#__PURE__*/_react.default.createElement(_button.ButtonStrip, {
    end: true
  }, /*#__PURE__*/_react.default.createElement(_button.Button, {
    type: "button",
    secondary: true,
    onClick: onClose
  }, _index.default.t('Close')))));
};

exports.Modal = Modal;
Modal.propTypes = {
  children: _propTypes.default.node.isRequired,
  name: _propTypes.default.string,
  onClose: _propTypes.default.func
};
