// @dhis2/ui-forms/build/cjs/SingleSelectFieldFF/SingleSelectFieldFF.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SingleSelectFieldFF = void 0;

var _select = require("@dhis2-ui/select");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _helpers = require("../shared/helpers.js");

var _propTypes2 = require("../shared/propTypes.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const SingleSelectFieldFF = _ref => {
  let {
    error,
    input,
    loading,
    meta,
    onBlur,
    onFocus,
    options,
    showLoadingStatus,
    showValidStatus,
    valid,
    validationText,
    ...rest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_select.SingleSelectField, _extends({}, rest, {
    name: input.name,
    error: (0, _helpers.hasError)(meta, error),
    valid: (0, _helpers.isValid)(meta, valid, showValidStatus),
    loading: (0, _helpers.isLoading)(meta, loading, showLoadingStatus),
    validationText: (0, _helpers.getValidationText)(meta, validationText, error),
    onFocus: (0, _helpers.createFocusHandler)(input, onFocus),
    onChange: (0, _helpers.createSelectChangeHandler)(input),
    onBlur: (0, _helpers.createBlurHandler)(input, onBlur),
    selected: input.value || ''
  }), options.map(option => /*#__PURE__*/_react.default.createElement(_select.SingleSelectOption, _extends({
    key: option.value
  }, option))));
};

exports.SingleSelectFieldFF = SingleSelectFieldFF;
SingleSelectFieldFF.propTypes = {
  /** `input` props received from Final Form `Field` */
  input: _propTypes2.inputPropType.isRequired,

  /** `meta` props received from Final Form `Field` */
  meta: _propTypes2.metaPropType.isRequired,
  options: _propTypes.default.arrayOf(_propTypes.default.shape({
    label: _propTypes.default.string,
    value: _propTypes.default.string
  })).isRequired,
  error: _propTypes.default.bool,
  loading: _propTypes.default.bool,
  showLoadingStatus: _propTypes.default.bool,
  showValidStatus: _propTypes.default.bool,
  valid: _propTypes.default.bool,
  validationText: _propTypes.default.string,
  onBlur: _propTypes.default.func,
  onFocus: _propTypes.default.func
};
