// @dhis2-ui/card/build/cjs/card.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;

var _box = require("@dhis2-ui/box");

var _react = _interopRequireDefault(require("react"));

var _card = require("./card.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const subtitle = "\nA card is a container element for grouping together\nand separating blocks of content.\n";
const description = "\nUse a card where there is content that can be grouped together.\nCards are most often useful when this grouped content may be repeated,\nfor example with items on a dashboard, or different sections of patient\ninformation displayed in a profile.\n\nNote that it requires a parent, like [Box](../?path=/docs/layout-box--default), to define its size.\n\n```js\nimport { Card } from '@dhis2/ui'\n```\n";
var _default = {
  title: 'Card',
  component: _card.Card,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  }
};
exports.default = _default;

const Default = args => /*#__PURE__*/_react.default.createElement(_box.Box, {
  width: "358px",
  height: "358px"
}, /*#__PURE__*/_react.default.createElement(_card.Card, args));

exports.Default = Default;
