// @dhis2-ui/transfer/build/cjs/end-intersection-detector.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EndIntersectionDetector = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _intersectionDetector = require("@dhis2-ui/intersection-detector");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const EndIntersectionDetector = _ref => {
  let {
    rootRef,
    onEndReached,
    dataTest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    className: "jsx-4200578429"
  }, /*#__PURE__*/_react.default.createElement(_intersectionDetector.IntersectionDetector, {
    rootRef: rootRef,
    onChange: _ref2 => {
      let {
        isIntersecting
      } = _ref2;
      return isIntersecting && onEndReached();
    }
  }), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "4200578429"
  }, ["div.jsx-4200578429{width:100%;height:50px;position:absolute;z-index:-1;bottom:0;left:0;}"]));
};

exports.EndIntersectionDetector = EndIntersectionDetector;
EndIntersectionDetector.propTypes = {
  rootRef: _propTypes.default.shape({
    current: _propTypes.default.instanceOf(HTMLElement)
  }).isRequired,
  onEndReached: _propTypes.default.func.isRequired,
  dataTest: _propTypes.default.string
};
