// @dhis2-ui/transfer/build/cjs/options-container.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OptionsContainer = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _loader = require("@dhis2-ui/loader");

var _uiConstants = require("@dhis2/ui-constants");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _endIntersectionDetector = require("./end-intersection-detector.js");

var _useResizeCounter = require("./use-resize-counter.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const OptionsContainer = _ref => {
  let {
    dataTest,
    emptyComponent,
    onEndReached,
    getOptionClickHandlers,
    highlightedOptions,
    loading,
    renderOption,
    options,
    selected,
    selectionHandler,
    toggleHighlightedOption
  } = _ref;
  const optionsRef = (0, _react.useRef)(null);
  const wrapperRef = (0, _react.useRef)(null);
  const resizeCounter = (0, _useResizeCounter.useResizeCounter)(wrapperRef.current);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["889751462", [_uiConstants.spacers.dp4]]]) + " " + "optionsContainer"
  }, loading && /*#__PURE__*/_react.default.createElement("div", {
    className: _style.default.dynamic([["889751462", [_uiConstants.spacers.dp4]]]) + " " + "loading"
  }, /*#__PURE__*/_react.default.createElement(_loader.CircularLoader, null)), /*#__PURE__*/_react.default.createElement("div", {
    "data-test": dataTest,
    ref: optionsRef,
    className: _style.default.dynamic([["889751462", [_uiConstants.spacers.dp4]]]) + " " + "container"
  }, /*#__PURE__*/_react.default.createElement("div", {
    ref: wrapperRef,
    className: _style.default.dynamic([["889751462", [_uiConstants.spacers.dp4]]]) + " " + "content-container"
  }, !options.length && emptyComponent, options.map(option => {
    const highlighted = !!highlightedOptions.find(highlightedSourceOption => highlightedSourceOption === option.value);
    return /*#__PURE__*/_react.default.createElement(_react.Fragment, {
      key: option.value
    }, renderOption({ ...option,
      ...getOptionClickHandlers(option, selectionHandler, toggleHighlightedOption),
      highlighted,
      selected
    }));
  }), onEndReached && /*#__PURE__*/_react.default.createElement(_endIntersectionDetector.EndIntersectionDetector, {
    dataTest: "".concat(dataTest, "-endintersectiondetector"),
    key: "key-".concat(resizeCounter),
    rootRef: optionsRef,
    onEndReached: onEndReached
  }))), /*#__PURE__*/_react.default.createElement(_style.default, {
    id: "889751462",
    dynamic: [_uiConstants.spacers.dp4]
  }, [".optionsContainer.__jsx-style-dynamic-selector{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding:".concat(_uiConstants.spacers.dp4, " 0;position:relative;overflow:hidden;}"), ".container.__jsx-style-dynamic-selector{overflow-y:auto;height:100%;}", ".loading.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;height:100%;width:100%;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;position:absolute;z-index:2;top:0;left:0;}", ".content-container.__jsx-style-dynamic-selector{z-index:1;position:relative;}", ".loading.__jsx-style-dynamic-selector+.container.__jsx-style-dynamic-selector .content-container.__jsx-style-dynamic-selector{-webkit-filter:blur(2px);filter:blur(2px);}"]));
};

exports.OptionsContainer = OptionsContainer;
OptionsContainer.defaultProps = {
  selected: false
};
OptionsContainer.propTypes = {
  dataTest: _propTypes.default.string.isRequired,
  getOptionClickHandlers: _propTypes.default.func.isRequired,
  emptyComponent: _propTypes.default.node,
  highlightedOptions: _propTypes.default.arrayOf(_propTypes.default.string),
  loading: _propTypes.default.bool,
  options: _propTypes.default.arrayOf(_propTypes.default.shape({
    label: _propTypes.default.string.isRequired,
    value: _propTypes.default.string.isRequired
  })),
  renderOption: _propTypes.default.func,
  selected: _propTypes.default.bool,
  selectionHandler: _propTypes.default.func,
  toggleHighlightedOption: _propTypes.default.func,
  onEndReached: _propTypes.default.func
};
