// @dhis2-ui/pagination/build/cjs/pagination.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithoutPageSummary = exports.WithoutPageSizeSelect = exports.WithoutGoToPageSelect = exports.WithoutAnySelect = exports.WithCustomPageSummary = exports.PagerAtLastPage = exports.PagerAtFirstPage = exports.NoTotalAtLastPageWithoutPageLength = exports.NoTotalAtLastPage = exports.NoTotal = exports.Disabled = exports.Default = void 0;

var _d2I18n = _interopRequireDefault(require("@dhis2/d2-i18n"));

var _react = _interopRequireDefault(require("react"));

var pagers = _interopRequireWildcard(require("./__fixtures__/index.js"));

var _pagination = require("./pagination.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const subtitle = 'Allows navigation through data displayed over several pages';
const description = "\nPagination allows data to be split in pages. Paging large amounts of data avoids overwhelming users and should always be used wherever a lot of data is displayed. Pagination controls allow a user to browse through a set of data or navigate to a specific page depending on the type of pagination used.\n\n**Do not rely on pagination for navigating datasets. A user should be able to search within, sort and filter datasets too, rather than needing to click through many pages looking for the right data item.**\n\n```js\nimport { Pagination } from '@dhis2/ui'\n```\n\n_**Note**: Dropdown menus may not display properly on this page. View these demos in the 'Canvas' tab._\n";

const logOnPageChange = page => {
  console.log("Now navigate to page ".concat(page, "..."));
};

const logOnPageSizeChange = pageSize => {
  console.log("Now change page size to ".concat(pageSize, "..."));
};

var _default = {
  title: 'Pagination',
  component: _pagination.Pagination,
  parameters: {
    componentSubtitle: subtitle,
    docs: {
      description: {
        component: description
      }
    }
  },
  // Default args for stories
  args: {
    onPageChange: logOnPageChange,
    onPageSizeChange: logOnPageSizeChange
  }
};
exports.default = _default;

const Template = args => /*#__PURE__*/_react.default.createElement(_pagination.Pagination, args);

const Default = Template.bind({});
exports.Default = Default;
Default.args = { ...pagers.atTenthPage
};
const Disabled = Template.bind({});
exports.Disabled = Disabled;
Disabled.args = { ...pagers.atTenthPage,
  disabled: true
};
const PagerAtFirstPage = Template.bind({});
exports.PagerAtFirstPage = PagerAtFirstPage;
PagerAtFirstPage.args = { ...pagers.atFirstPage
};
const PagerAtLastPage = Template.bind({});
exports.PagerAtLastPage = PagerAtLastPage;
PagerAtLastPage.args = { ...pagers.atLastPage
};
const NoTotal = Template.bind({});
exports.NoTotal = NoTotal;
NoTotal.args = { ...pagers.noTotal
};
const NoTotalAtLastPage = Template.bind({});
exports.NoTotalAtLastPage = NoTotalAtLastPage;
NoTotalAtLastPage.args = { ...pagers.noTotalAtLastPage,
  pageLength: 26
};
const NoTotalAtLastPageWithoutPageLength = Template.bind({});
exports.NoTotalAtLastPageWithoutPageLength = NoTotalAtLastPageWithoutPageLength;
NoTotalAtLastPageWithoutPageLength.args = { ...pagers.noTotalAtLastPage
};
const WithoutPageSizeSelect = Template.bind({});
exports.WithoutPageSizeSelect = WithoutPageSizeSelect;
WithoutPageSizeSelect.args = { ...pagers.atTenthPage,
  hidePageSizeSelect: true
};
const WithoutGoToPageSelect = Template.bind({});
exports.WithoutGoToPageSelect = WithoutGoToPageSelect;
WithoutGoToPageSelect.args = { ...pagers.atTenthPage,
  hidePageSelect: true
};
const WithoutPageSummary = Template.bind({});
exports.WithoutPageSummary = WithoutPageSummary;
WithoutPageSummary.args = { ...pagers.atTenthPage,
  hidePageSummary: true
};
const WithCustomPageSummary = Template.bind({});
exports.WithCustomPageSummary = WithCustomPageSummary;
WithCustomPageSummary.args = { ...pagers.atTenthPage,
  pageSummaryText: interpolationObject => _d2I18n.default.t('You are at page {{page}} showing items {{firstItem}}-{{lastItem}}, but there are {{pageCount}} pages and {{total}} items', interpolationObject)
};
const WithoutAnySelect = Template.bind({});
exports.WithoutAnySelect = WithoutAnySelect;
WithoutAnySelect.args = { ...pagers.atTenthPage,
  ...WithoutGoToPageSelect.args,
  ...WithoutPageSizeSelect.args
};
